var/spack/repos/builtin/packages/py-cdo/package.py

```diff
@@ -12,10 +12,21 @@ class PyCdo(PythonPackage):
 
     pypi = "cdo/cdo-1.3.2.tar.gz"
 
-    version("1.3.2", sha256="9f78879d90d14134f2320565016d0d371b7dfe7ec71110fd313868ec1db34aee")
+    maintainers = ["Try2Code", "skosukhin"]
 
-    depends_on("cdo")
+    version("1.5.6", sha256="fec1a75382f01b3c9c368e8f143d98b12323e06975663f87d9b60c739ae1d335")
+    version(
+        "1.3.2",
+        sha256="9f78879d90d14134f2320565016d0d371b7dfe7ec71110fd313868ec1db34aee",
+        deprecated=True,
+    )
 
     depends_on("py-setuptools", type="build")
-    depends_on("py-scipy", type=("build", "run"))
+    depends_on("cdo+netcdf", type="run")
     depends_on("py-netcdf4", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"), when="@:1.4")
+    depends_on("py-xarray", type=("build", "run"), when="@1.3.4:")
```

Comment:
Seems like netcdf4 and xarray are optional, not sure if we should add variants for these or always add them

Reply:
I recommend to always add them. From my experience most users have workflows, which involve netcdf4 files and XArray. Adding those dependencies just makes the package more user friendly IMO.